06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js

```diff
@@ -0,0 +1,174 @@
+const Index = require('../Model/inverted-index.js');
+const invalidBook = require('../files/test.json');
+const workBook = require('../files/work.json');
+const empty = require('../files/empty.json');
+
+const invertedIndex = new Index();
+
+const work = [{
+  title: 'Checkpoint one',
+  text: 'Testing getIndex function',
+},
+
+{
+  title: 'Checkpoint Two',
+  text: 'Coming soon',
+},
+];
+
+describe('Define methods for Index class', () => {
+  invertedIndex.createIndex('work.json', workBook);
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.getIndex).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.createIndex).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.tokenize).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(Index.validateFile).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.searchIndex).toBeDefined();
+  });
+});
+
+describe('Should return an object that is an index of the file', () => {
+  invertedIndex.createIndex('work.json', work);
+  it('should verify that index has been created', () => {
+    expect(Object.keys(invertedIndex.getIndex('work.json', work)).length)
+      .toBeGreaterThan(0);
+  });
```

Comment:
I was actually testing to make sure the methods are function also, but I have made changes by removing the the block of code since it's not required